packages/wallet-connect-sdk-react/src/index.tsx

```diff
@@ -5,43 +5,230 @@ import {ContractInvocation, ContractInvocationMulti, RpcCallResult, WcSdk} from
 
+/**
+ * WalletConnect's context for React
+ */
 interface IWalletConnectContext {
+    /**
+     * Wallet Connect Client, used on most of the methods
+     */
     wcClient: Client | undefined,
     setWcClient: React.Dispatch<React.SetStateAction<Client | undefined>>,
+
+    /**
+     * The Session of the current connection
+     */
     session: SessionTypes.Created | undefined,
     setSession: React.Dispatch<React.SetStateAction<SessionTypes.Created | undefined>>,
+
+    /**
+     * An indicator if the session is loading
+     */
     loadingSession: boolean,
     setLoadingSession: React.Dispatch<React.SetStateAction<boolean>>,
-    pairings: string[],
-    setPairings: React.Dispatch<React.SetStateAction<string[]>>,
-    isPairing: boolean,
-    setIsPairing: React.Dispatch<React.SetStateAction<boolean>>,
+
+    /**
+     * An indicator for when there is any request that is pending for the wallet to approve, use this variable to ask the users to check their wallet to approve the request
+     */
     isPendingApproval: boolean,
     setIsPendingApproval: React.Dispatch<React.SetStateAction<boolean>>,
+
+    /**
+     * The Connection Proposal URI, use this variable to show a custom QRCode
+     *
+     * Note: The QRCodeModal is automatically shown by default, to disable this feature set `qrCodeModal` option as false
+     *
+     * ```
+     * useEffect(() => {
+     *     if (walletConnectCtx.uri.length) {
+     *         window.open(`https://neon.coz.io/connect?uri=${walletConnectCtx.uri}`, '_blank')?.focus();
+     *     }
+     * }, [walletConnectCtx.uri])
+     * ```
+     */
     uri: string,
     setUri: React.Dispatch<React.SetStateAction<string>>,
+
+    /**
+     * Connected Accounts, the string is composed by the format: `${chainName}:${chainEnvironment}:${accountAddress}`
+     *
+     * Use `walletConnectCtx.getAccountAddress()` to retrieve only the accountAddress
+     */
     accounts: string[],
     setAccounts: React.Dispatch<React.SetStateAction<string[]>>,
 
-    openPairing: () => Promise<void>,
+    /**
+     * Initialize the connection process
+     * @param topic optional pairing topic to reestablish an existing connection
+     */
     connect: (topic?: string) => Promise<void>,
+
+    /**
+     * Sends a request to the Wallet and it will call the RpcServer
+     * ```
+     * const result = await walletConnectCtx.sendRequest({
+     *   method: 'getapplicationlog',
+     *   params: ['0x7da6ae7ff9d0b7af3d32f3a2feb2aa96c2a27ef8b651f9a132cfaad6ef20724c']
+     * })
+     * ```
+     * @param request the request information object containing the rpc method name and the parameters
+     * @return the call result promise
+     */
     sendRequest: (request: RequestArguments) => Promise<RpcCallResult>,
+
+    /**
+     * Sends an 'invokefunction' request to the Wallet and it will communicate with the blockchain. It will consume gas and persist data to the blockchain.
```

Comment:
```suggestion
     * Sends an 'invokefunction' request to the Wallet, which should approve it and communicate the transaction to the blockchain. It will consume gas and persist data to the blockchain.

```